From 626da7724e55cb637c4576d6166a130df0791efb Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Thu, 10 Jul 2025 12:57:27 -0700
Subject: [PATCH] debugging hack to show diff when kallsyms cmp fails

powerpc allmodconfig has been failing on tuxsuite but it is not
reproducible for me locally. Print the diff of the files to see if
anything obvious sticks out.
---
 scripts/link-vmlinux.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/link-vmlinux.sh b/scripts/link-vmlinux.sh
index 51367c2bfc21..4773b9c0d198 100755
--- a/scripts/link-vmlinux.sh
+++ b/scripts/link-vmlinux.sh
@@ -309,6 +309,7 @@ if is_enabled CONFIG_KALLSYMS; then
 	if ! cmp -s System.map "${kallsyms_sysmap}"; then
 		echo >&2 Inconsistent kallsyms data
 		echo >&2 'Try "make KALLSYMS_EXTRA_PASS=1" as a workaround'
+		git diff --no-index System.map "${kallsyms_sysmap}" >&2
 		exit 1
 	fi
 fi
-- 
2.50.1

